File: Algorithm/Sorting Algorithm/selection_sort/src/selection_sort.cpp
/**
 * @file selection_sort.cpp
 * @author Sumit Kumar (ravirajkarn@outlook.com)
 * @brief This is binary search algorithm created my me.
 * @version 0.1
 * @date 2026-05-20
 *
 * @copyright Copyright (c) 2026
 *
 */

#include <selection_sort.hpp>

int main() {
  std::vector<int> number = {64, 25, 12, 22, 11, -1, 0, 3, 5, 9, 12};
  std::print("the numbe befor: {}\n", number);
  selectionSort(number);
  std::print("the numbe after: {}\n", number);
  return 0;
}

File: Algorithm/Sorting Algorithm/selection_sort/src/selection_sort.py
# This code is created by me
def selection_sort(nums):
    if len(nums) < 1:
        return nums
    for index in range(len(nums)):
        selected = index
        selector = index
        for selector in range(index + 1, len(nums)):
            if nums[selector] < nums[selected]:
                selected = selector
        if index != selected:
            nums[index], nums[selected] = nums[selected], nums[index]


def main():
    number = [64, 25, 12, 22, 11, -1, 0, 3, 5, 9, 12]
    selection_sort(number)
    print(f"the number is: {number}")


if __name__ == "__main__":
    main()
